Wrapping/Generators/SwigInterface/igenerator.py

```diff
@@ -585,13 +582,13 @@ def generate_enum(self, typedef):
         self.outputFile.write("%{\n")
         self.outputFile.write("using namespace %s;\n" % ns)
         self.outputFile.write("%}\n")
-        content = [" %s = %i" % (key, value) for key, value in enum.values]
-        self.outputFile.write("enum %s { %s };\n" % (name, ", ".join(content)))
+        content = [" %s" % (key,) for key, value in enum.values]
+        self.outputFile.write("enum class %s: uint8_t { %s };\n\n" % (name, ", ".join(content)))
 
     def generate_nested_enum(self, typedef, enum, indent, w):
-        content = [" %s = %i" % (key, value) for key, value in enum.values]
+        content = [" %s" % (key,) for key, value in enum.values]
         self.outputFile.write("  " * indent)
-        self.outputFile.write("    enum %s { %s };\n" % (enum.name, ", ".join(content)))
+        self.outputFile.write("    enum class %s: uint8_t { %s };\n\n" % (enum.name, ", ".join(content)))
```

Comment:
AHHH!  This is the magic fix that I could not track down!  I could not figure out where I was going wrong. 

I hope this solution works.

Reply:
It is great to have more folks understanding this process. I did not know how enum's worked in the wrapping process previously, but now appears to be quite simple.

In `igenerator.py`, we create the SWIG interface *.i* files that define the Python interface. We have a CastXML data structure that represents the abstract syntax tree. We take the enum's and tell SWIG to generate the interface for strongly typed enum's.